mmv1/products/dlp/JobTrigger.yaml

```diff
@@ -158,6 +175,413 @@ properties:
+
+              When no InfoTypes or CustomInfoTypes are specified in a request, the system may automatically choose what detectors to run.
+              By default this may be all types, but may change over time as detectors are updated.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::String
+                  name: 'name'
+                  required: true
+                  description: |
+                    Name of the information type. Either a name of your choosing when creating a CustomInfoType, or one of the names listed
+                    at https://cloud.google.com/dlp/docs/infotypes-reference when specifying a built-in type.
+                - !ruby/object:Api::Type::String
+                  name: 'version'
+                  description: |
+                    Version of the information type to use. By default, the version is set to stable
+          - !ruby/object:Api::Type::Array
+            name: 'ruleSet'
+            description: |
+              Set of rules to apply to the findings for this InspectConfig. Exclusion rules, contained in the set are executed in the end,
+              other rules are executed in the order they are specified for each info type.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Array
+                  name: 'infoTypes'
+                  required: true
+                  description: |
+                    List of infoTypes this rule set is applied to.
+                  item_type: !ruby/object:Api::Type::NestedObject
+                    properties:
+                      - !ruby/object:Api::Type::String
+                        name: 'name'
+                        required: true
+                        description: |
+                          Name of the information type. Either a name of your choosing when creating a CustomInfoType, or one of the names listed
+                          at https://cloud.google.com/dlp/docs/infotypes-reference when specifying a built-in type.
+                      - !ruby/object:Api::Type::String
+                        name: 'version'
+                        description: |
+                          Version of the information type to use. By default, the version is set to stable.
+                - !ruby/object:Api::Type::Array
+                  name: 'rules'
+                  required: true
+                  description: |
+                    Set of rules to be applied to infoTypes. The rules are applied in order.
+                  item_type: !ruby/object:Api::Type::NestedObject
+                    properties:
+                      - !ruby/object:Api::Type::NestedObject
+                        name: 'hotwordRule'
+                        description: Hotword-based detection rule.
+                        properties:
+                          - !ruby/object:Api::Type::NestedObject
```

Comment:
Based on the API doc, https://cloud.google.com/dlp/docs/reference/rest/v2/InspectConfig#InspectionRule, Union field type can be only one of the following: hotwordRule and exclusionRule. exactly_one_of is needed to added to those two fields.

Reply:
Currently, `ConflictsWith`, `ExactlyOneOf` or `AtLeastOneOf` are set at the attribute level, which means we can't set it on a list if we have a list of objects, and each object has a list of attributes that's `AtLeastOneOf`. We don't know the length of the list and there is no way to differentiate object[0] from object[1] with what is conflicting. Hence, the use of the above mentioned functions when the attributes are part of array won't work.